locust/contrib/fasthttp.py

```diff
@@ -0,0 +1,383 @@
+        :param path: Path that will be concatenated with the base host URL that has been specified.
+            Can also be a full URL, in which case the full URL will be requested, and the base host 
+            is ignored.
+        :param name: (optional) An argument that can be specified to use as label in Locust's 
+            statistics instead of the URL path. This can be used to group different URL's 
+            that are requested into a single entry in Locust's statistics.
+        :param catch_response: (optional) Boolean argument that, if set, can be used to make a request 
+            return a context manager to work as argument to a with statement. This will allow the 
+            request to be marked as a fail based on the content of the response, even if the response 
+            code is ok (2xx). The opposite also works, one can use catch_response to catch a request 
+            and then mark it as successful even if the response code was not (i.e 500 or 404).
+        :param data: (optional) Dictionary or bytes to send in the body of the request.
+        :param headers: (optional) Dictionary of HTTP Headers to send with the request.
+        :param auth: (optional) Auth (username, password) tuple to enable Basic HTTP Auth.
+        :param stream: (optional) If set to true the response body will not be consumed immediately 
+            and can instead be consumed by accessing the stream attribute on the Response object.
+            Another side effect of setting stream to True is that the time for downloading the response 
+            content will not be accounted for in the request time that is reported by Locust.
+        """
+        # prepend url with hostname unless it's already an absolute URL
+        url = self._build_url(path)
+        
+        # store meta data that is used when reporting the request to locust's statistics
+        request_meta = {}
+        # set up pre_request hook for attaching meta data to the request object
+        request_meta["method"] = method
+        request_meta["start_time"] = default_timer()
+        request_meta["name"] = name or path
+        
+        if auth:
+            headers = headers or {}
+            headers['Authorization'] = _construct_basic_auth_str(auth[0], auth[1])
+        elif self.auth_header:
+            headers = headers or {}
+            headers['Authorization'] = self.auth_header
+        
+        # send request, and catch any exceptions
+        response = self._send_request_safe_mode(method, url, payload=data, headers=headers, **kwargs)
+        
+        # get the length of the content, but if the argument stream is set to True, we take
+        # the size from the content-length header, in order to not trigger fetching of the body
+        if stream:
+            request_meta["content_size"] = int(response.headers.get("content-length") or 0)
+        else:
+            request_meta["content_size"] = len(response.content or "")
```

Comment:
In my testing of this, I ran into an issue with a response's content not being saved properly.

When using `FastHttpSession` and making a request, if `stream=False` the response always returns with `.content` being `bytearray(b'')`. If `stream=True`, the content is as expected. I'm not sure, but from what I can see the problem seems to stem from here where the content is read into the `len()` call which then causes the urllib response to release `content`.

I'm currently working around this by adding `request_meta["content"] = response.content` before this `if stream` check, but I'm sure there's a better way to retain content.